data/blog/2025/2025-01-30-mezzio101.md

```diff
@@ -0,0 +1,174 @@
+
+## What is middleware?
+
+Middleware is code that is executed between the request and response.
+Most commonly, it performs these tasks, but more layers can be defined based on your application:
+
+- Aggregates incoming data.
+- Processes the request.
+- Creates and returns a response.
+
+There are several advantages to using middleware:
+
+- It enables old systems to interface with newer ones.
+- It's scalable.
+- Integrations are created more easily.
+- It enables automating processes.
+
+## Installing Mezzio
+
+Let's put aside the technical stuff in favor of some real work that you can be proud of.
+Building things with our hands, kind of.
+
+To start, create a folder where you want your project files to be created.
+
+```shell
+mkdir -p ~/Projects/my-mezzio-project
+```
+
+The next command will launch the Mezzio Skeleton Installer.
+We will go through the installation process together, using the recommended settings.
+We will opt primarily for the Laminas-supported options to maintain consistency.
+Funnily enough, the installation should complete in less time than it takes for you to read this tutorial.
+
+```shell
+composer create-project mezzio/mezzio-skeleton ~/Projects/my-mezzio-project
+```
+
+You should see the text and prompt below:
+
+![step-1-installation-type](/images/blog/mezzio101/mezzio101-use-skeleton-01.jpg "step-1")
+
+Submit `3` for option `Modular`.
+
+![step-2-dependency-injection](/images/blog/mezzio101/mezzio101-use-skeleton-02.jpg "step-2")
+
+Submit `1` for option `laminas-servicemanager`.
+Just as a side note, this option is Laminas's factory-driven dependency injection container.
+Its version 4.0 supports PSR-11: Container interface version 1.1 and 2.
+
+![step-3-router](/images/blog/mezzio101/mezzio101-use-skeleton-03.jpg "step-3")
+
+Submit `1` for `fastroute`.
+
+![step-4-templating](/images/blog/mezzio101/mezzio101-use-skeleton-04.jpg "step-4")
+
+Submit `2` for `Twig`.
+
+![step-5-error-reporting](/images/blog/mezzio101/mezzio101-use-skeleton-05.jpg "step-5")
+
+Submit `1` for `Whoops`.
+At this point, the installer has enough information to install the 3rd party packages in the application.
+
+You can expect to see this prompt regarding dependency injection.
+
+![step-6-injection](/images/blog/mezzio101/mezzio101-use-skeleton-06.jpg "step-6")
+
+Submit `1` for `config/config.php`.
+
+![step-7-injection-confirmation](/images/blog/mezzio101/mezzio101-use-skeleton-07.jpg "step-7")
+
+Submit `y` for `Remember this option for other packages of the same type? (Y/n)`.
+
+The final step in the installation performs the following operations:
+
+- Sets up `phpcs` the [PHP Code Sniffer](https://github.com/PHPCSStandards/PHP_CodeSniffer)
+- Clears the cache
+- Enables development mode
+
+![step-8-installation-complete](/images/blog/mezzio101/mezzio101-use-skeleton-08.jpg "step-8")
+
+### The result of your work so far
+
```

Comment:
I expanded this section to make sure the reader has the steps to review the installation result